src/RequestParser.cpp

```diff
@@ -0,0 +1,724 @@
+ * @param request The HTTP request object to be filled.
+ * @throws std::runtime_error If the chunked body format is invalid (missing CRLF or incorrect chunk size).
+ *
+ * Error codes:
+ * - ERR_MISS_CRLF: Thrown when a line does not end with a CRLF.
+ * - ERR_CHUNK_SIZE: Thrown when the chunk size does not match the specified size.
+ *
+ * Example usage:
+ * @code
+ * std::istringstream requestStream("4\r\nWiki\r\n5\r\npedia\r\n0\r\n\r\n");
+ * RequestParser parser;
+ * parser.parseChunkedBody(requestStream);
+ * @endcode
+ */
+void RequestParser::parseChunkedBody(HTTPRequest& request)
+{
+	size_t length = 0;
+	std::string strChunkSize;
+	std::string chunkData;
+	size_t numChunkSize = 0;
+
+	do {
+		std::getline(m_requestStream, strChunkSize);
+		if (strChunkSize[strChunkSize.size() - 1] == '\r')
+			strChunkSize.erase(strChunkSize.size() - 1);
+		else {
+			request.errorCode = StatusBadRequest;
+			throw std::runtime_error(ERR_MISS_CRLF);
+		}
+		numChunkSize = convertHex(strChunkSize);
+		std::getline(m_requestStream, chunkData);
+		if (chunkData[chunkData.size() - 1] == '\r')
+			chunkData.erase(chunkData.size() - 1);
+		else {
+			request.errorCode = StatusBadRequest;
+			throw std::runtime_error(ERR_MISS_CRLF);
+		}
+		if (chunkData.size() != numChunkSize) {
+			request.errorCode = StatusBadRequest;
+			throw std::runtime_error(ERR_CHUNK_SIZE);
+		}
+		request.body += chunkData;
+		length += numChunkSize;
+	} while (numChunkSize > 0);
+}
+
+/**
+ * @brief Parses a non-chunked body from the provided input stream.
+ *
+ * This function reads the entire body from the input stream, ensuring that the
+ * total length of the body matches the "Content-Length" header specified in the request.
+ * It processes each line, removing trailing carriage returns and concatenates
+ * the lines to form the complete body.
+ *
+ * @param request The HTTP request object to be filled.
+ * @throws std::runtime_error If there is an error converting the "Content-Length" header
+ *                            to a size_t or if the body length does not match the "Content-Length" value.
+ *
+ * Error codes:
+ * - ERR_CONVERSION_STRING_TO_SIZE_T: Thrown when the conversion of "Content-Length" header to size_t fails.
+ * - ERR_CONTENT_LENGTH: Thrown when the length of the parsed body does not match the "Content-Length" value.
+ *
+ * Example usage:
+ * @code
+ * std::istringstream requestStream("This is the body of the request.\r\n");
+ * RequestParser parser;
+ * parser.parseNonChunkedBody(requestStream);
+ * @endcode
+ */
+void RequestParser::parseNonChunkedBody(HTTPRequest& request)
+{
+	std::string body;
+	size_t length = 0;
+
+	while (std::getline(m_requestStream, body)) {
+		if (body[body.size() - 1] == '\r') {
+			body.erase(body.size() - 1);
+			length += 1;
+		}
+		if (!m_requestStream.eof())
+			body += '\n';
+		length += body.size();
+		request.body += body;
+	}
+	size_t contentLength = strtol(request.headers.at("Content-Length").c_str(), NULL, decimalBase);
```

Comment:
strtol returns long, implicit conversion to unsigned long (size_t) --> not caught by clang tidy?
Could use std namespace version: std::strtol
"long" contentLength can be made const